Add user docs for molecular and periodic DF (#17)
pyscf/pyscf.github.io: source/user/pbc/df.rst

Density fitting for crystalline calculations
********************************************

*Modules*: :mod:`df`, :mod:`pbc.df`

Introduction
============

> Like for molecular calculations, density fitting (DF) is useful to reduce the computational cost of manipulating the electron repulsion integrals (ERIs) in periodic calculations. Currently, PySCF provides three periodic DF methods that are different in the auxiliary basis functions being used.

Reviewer: ```suggestion
In addition to molecular calculations, density fitting (DF) is also useful in periodic calculations, as it reduces the computational cost of manipulating the electron repulsion integrals (ERIs).
Currently, PySCF provides three periodic DF methods that differ in their definition for their auxiliary basis functions.
```

Author: fixed

* The fast Fourier transform density fitting (:class:`FFTDF`) uses plane waves (PWs) as the auxiliary basis functions. FFTDF is also known as the Gaussian and plane wave (GPW) approach in literature and other software packages. :cite:`VandeVondele05CPC,Kuhne20JCP`

> * The Gaussian density fitting (:class:`GDF`) uses Gaussian-type orbitals (GTOs) as the auxiliary basis functions. :cite:`Sun17JCP`

Reviewer: I think this should be the first item

Author: I put FFTDF first bc it's the *default* used by the periodic SCF module (i.e., the default is not "without DF" but FFTDF). What do you think?

* The mixed density fitting (:class:`MDF`) uses both PWs and GTOs as the auxiliary basis functions. :cite:`Sun17JCP`

This section of the user document covers the basic usage of these DF methods in periodic calculations. The recommended choice of DF methods for different applications is also provided. See :ref:`theory_df` for the use of DF in molecular calculations.

Using DF in periodic calculations
=================================

> Unlike for molecular calculations, DF is used by default in periodic SCF calculations. For example, initialize a KRHF object uses FFTDF by default::

Reviewer: what is KRHF?

Reviewer: RHF with k-point sampling?

Reviewer: You should write it out

    import numpy as np
    from pyscf.pbc import gto, scf
    cell = gto.M(atom="He 0 0 0", a=np.eye(3)*2, basis="6-31g")
    kpts = cell.make_kpts([2,2,2])
    mf = scf.KRHF(cell, kpts)
    print(mf.with_df)  # <pyscf.pbc.df.fft.FFTDF at 0x7feb78373c88>

The periodic SCF class also provides methods for using GDF::

    mf = scf.KRHF(cell, kpts).density_fit()

and MDF::

    mf = scf.KRHF(cell, kpts).mix_density_fit()

In addition to using the APIs above, the user can also initialize a DF method first and overwrite the attribute :attr:`with_df` of the SCF object. For example::

    >>> from pyscf.pbc import df
    >>> mydf = df.GDF(cell, kpts).build()
    >>> mf.with_df = mydf

> Once the periodic mean-field calculation is done, the subsequent correlated calculations will automatically use the same DF method to handle ERIs. For example, a MP2 calculation using GDF can be performed as follows::

Reviewer: Same note here as above for the molecular one
